java/tables/src/main/java/sleeper/table/util/StateStoreProvider.java

```diff
@@ -25,25 +25,39 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.function.Function;
 
 import static sleeper.configuration.properties.table.TableProperty.TABLE_NAME;
 
 public class StateStoreProvider {
-    private final StateStoreFactory stateStoreFactory;
+    private final Function<TableProperties, StateStore> stateStoreFactory;
     private final Map<String, StateStore> tableNameToStateStoreCache;
 
     public StateStoreProvider(AmazonDynamoDB dynamoDBClient,
                               InstanceProperties instanceProperties,
                               Configuration configuration) {
-        this.stateStoreFactory = new StateStoreFactory(dynamoDBClient, instanceProperties, configuration);
-        this.tableNameToStateStoreCache = new HashMap<>();
+        this(new StateStoreFactory(dynamoDBClient, instanceProperties, configuration)::getStateStore);
     }
 
     public StateStoreProvider(AmazonDynamoDB dynamoDBClient,
                               InstanceProperties instanceProperties) {
         this(dynamoDBClient, instanceProperties, null);
     }
 
+    private StateStoreProvider(Function<TableProperties, StateStore> stateStoreFactory) {
+        this.stateStoreFactory = stateStoreFactory;
+        this.tableNameToStateStoreCache = new HashMap<>();
+    }
+
+    public static StateStoreProvider fixed(TableProperties singleTableProperties, StateStore stateStore) {
+        return new StateStoreProvider(tableProperties -> {
+            if (tableProperties != singleTableProperties) {
+                throw new IllegalArgumentException("Table not found: " + tableProperties.get(TABLE_NAME));
+            }
+            return stateStore;
+        });
+    }
+
```

Comment:
This method looks like it would only ever be useful for tests? Could it go in a class in src/tests, rather than src/main?

Reply:
I'm not sure I'm happy with this, it makes it quite a bit harder to find. Note that this is also an issue with TablePropertiesProvider.

Reply:
Good point about TablePropertiesProvider. We should be consistent. I don't see that it being harder to find is that big a deal. Maybe we need a module for all generally useful test code?

Reply:
If we had a general test code module it'd probably need a lot of dependencies, which might get confusing.

I've added FixedStateStoreProvider and FixedTablePropertiesProvider in the test code for the same module as their non-fixed classes. I also moved StateStoreProvider to the statestore module.